src/codegen.cpp

```diff
@@ -3200,14 +3204,16 @@ static jl_cgval_t emit_expr(jl_value_t *expr, jl_codectx_t *ctx)
         // this way until a later optimization pass, so it might see one of our
         // SSA vars as not dominating all uses. see issue #6068
         // Work around this by generating unconditional branches.
-        if (cond == jl_true) {
+        jl_cgval_t condV = emit_expr(cond, ctx);
+        jl_value_t *static_cond = static_eval(cond, ctx, true, true);
```

Comment:
Yeah, this is not needed anymore. This was the first solution I tried before realizing the `Const` nodes are widened/striped from the AST before the type inference is done....

Deleted in https://github.com/JuliaLang/julia/pull/16341